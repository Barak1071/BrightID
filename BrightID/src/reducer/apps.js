// @flow

import { SET_APPS, ADD_LINK, RESET_STORE } from '@/actions';

const initialState = {
  apps: [],
  links: [],
};

export const reducer = (state: AppsState = initialState, action: action) => {
  switch (action.type) {
    case ADD_LINK: {
      const removeExisting = ({ context }) => context !== action.link.context;
      const links: LinkInfo[] = state.links
        .filter(removeExisting)
        .concat(action.link);
      return {
        ...state,
        links,
      };
    }
    case SET_APPS: {
      return {
        ...state,
        apps: action.apps,
      };
    }
    case RESET_STORE: {
      return { ...initialState };
    }
    default: {
      return state;
    }
  }
};

export default reducer;
