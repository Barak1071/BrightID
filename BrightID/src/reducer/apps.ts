import {
  SET_APPS,
  ADD_LINKED_CONTEXT,
  REMOVE_LINKED_CONTEXT,
  RESET_STORE,
} from '@/actions';

const initialState: AppsState = {
  apps: [],
  linkedContexts: [],
};

export const reducer = (state: AppsState = initialState, action: action) => {
  switch (action.type) {
    case ADD_LINKED_CONTEXT: {
      const removeExisting = ({ context }) => context !== action.link.context;
      const linkedContexts: ContextInfo[] = state.linkedContexts
        .filter(removeExisting)
        .concat(action.link);
      return {
        ...state,
        linkedContexts,
      };
    }
    case REMOVE_LINKED_CONTEXT: {
      const removeExisting = ({ context }) => context !== action.context;
      const linkedContexts: ContextInfo[] = state.linkedContexts.filter(
        removeExisting,
      );

      return { ...state, linkedContexts };
    }
    case SET_APPS: {
      return {
        ...state,
        apps: action.apps,
      };
    }
    case RESET_STORE: {
      return { ...initialState };
    }
    default: {
      return state;
    }
  }
};

export default reducer;
