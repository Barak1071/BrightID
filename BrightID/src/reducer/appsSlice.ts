import {
  createSelector,
  createSlice,
  createEntityAdapter,
  PayloadAction,
  Update,
} from '@reduxjs/toolkit';
import { RESET_STORE } from '@/actions/resetStore';
import { sponsoring_steps } from '@/utils/constants';

/* ******** INITIAL STATE ************** */

const linkedContextsAdapter = createEntityAdapter<ContextInfo>({
  selectId: (linkedContext) => linkedContext.contextId,
});

const sigsAdapter = createEntityAdapter<SigInfo>({
  selectId: (sig) => `${sig.uid}`,
});

const initialState: AppsState = {
  apps: [],
  linkedContexts: linkedContextsAdapter.getInitialState(),
  sigs: sigsAdapter.getInitialState(),
  sigsUpdating: false,
  sponsoringStep: sponsoring_steps.IDLE,
  sponsoringStepText: undefined,
  linkingAppInfo: undefined,
  linkingAppStartTime: 0,
  sponsorOperationHash: undefined,
};

const appsSlice = createSlice({
  name: 'apps',
  initialState,
  reducers: {
    setApps(state, action: PayloadAction<AppInfo[]>) {
      state.apps = action.payload;
    },
    addLinkedContext(state, action: PayloadAction<ContextInfo>) {
      state.linkedContexts = linkedContextsAdapter.addOne(
        state.linkedContexts,
        action,
      );
    },
    removeLinkedContext(state, action: PayloadAction<string>) {
      state.linkedContexts = linkedContextsAdapter.removeOne(
        state.linkedContexts,
        action,
      );
    },
    updateLinkedContext(state, action: PayloadAction<Partial<ContextInfo>>) {
      const update: Update<ContextInfo> = {
        id: action.payload.contextId,
        changes: action.payload,
      };
      state.linkedContexts = linkedContextsAdapter.updateOne(
        state.linkedContexts,
        update,
      );
    },
    upsertSig(state, action: PayloadAction<SigInfo>) {
      state.sigs = sigsAdapter.upsertOne(state.sigs, action);
    },
    removeSig(state, action: PayloadAction<string>) {
      state.sigs = sigsAdapter.removeOne(state.sigs, action);
    },
    removeAllSigs(state) {
      state.sigs = sigsAdapter.removeAll(state.sigs);
    },
    updateSig(state, action: PayloadAction<Update<SigInfo>>) {
      state.sigs = sigsAdapter.updateOne(state.sigs, action.payload);
    },
    setSigsUpdating(state, action: PayloadAction<boolean>) {
      state.sigsUpdating = action.payload;
    },
    setSponsoringStep(
      state,
      action: PayloadAction<{ step: SponsoringStep_Type; text?: string }>,
    ) {
      const { step, text } = action.payload;
      state.sponsoringStep = step;
      state.sponsoringStepText = text;
    },
    setLinkingAppInfo(
      state,
      action: PayloadAction<LinkingAppInfo | undefined>,
    ) {
      state.linkingAppInfo = action.payload;
    },
    setLinkingAppStarttime(state, action: PayloadAction<number>) {
      state.linkingAppStartTime = action.payload;
    },
    setSponsorOperationHash(state, action: PayloadAction<string>) {
      state.sponsorOperationHash = action.payload;
    },
    resetLinkingAppState(state) {
      state.sponsoringStep = sponsoring_steps.IDLE;
      state.sponsoringStepText = undefined;
      state.linkingAppInfo = undefined;
      state.linkingAppStartTime = 0;
      state.sponsorOperationHash = undefined;
    },
  },
  extraReducers: {
    [RESET_STORE]: () => {
      return initialState;
    },
  },
});

// Export app actions
export const {
  setApps,
  addLinkedContext,
  removeLinkedContext,
  updateLinkedContext,
  upsertSig,
  removeSig,
  removeAllSigs,
  updateSig,
  setSigsUpdating,
  setSponsoringStep,
  setLinkingAppInfo,
  setLinkingAppStarttime,
  setSponsorOperationHash,
  resetLinkingAppState,
} = appsSlice.actions;

export const {
  selectById: selectLinkedContextById,
  selectAll: selectAllLinkedContexts,
} = linkedContextsAdapter.getSelectors(
  (state: RootState) => state.apps.linkedContexts,
);

export const { selectAll: selectAllSigs } = sigsAdapter.getSelectors(
  (state: RootState) => state.apps.sigs,
);

export const linkedContextTotal = createSelector(
  selectAllLinkedContexts,
  (contexts) =>
    contexts.reduce(
      (acc, link) => (link.state === 'applied' ? acc + 1 : acc),
      0,
    ),
);

export const selectLinkedContext = createSelector(
  selectAllLinkedContexts,
  (_: RootState, context: string) => context,
  (contexts, context) => contexts.find((link) => link.context === context),
);

export const selectPendingLinkedContext = createSelector(
  selectAllLinkedContexts,
  (contexts) => contexts.find((link) => link.state === 'pending'),
);

export const selectAllApps = (state: RootState) => state.apps.apps;

export const selectAllLinkedSigs = createSelector(selectAllSigs, (sigs) =>
  sigs.filter((sig) => sig.linked),
);

export const createSelectLinkedSigsForApp = (appId) =>
  createSelector(selectAllLinkedSigs, (linkedSigs) => {
    // return all linked sigs that belong to provided app
    return linkedSigs.filter((sig) => sig.app === appId);
  });

export const selectBlindSigApps = (state: RootState) =>
  state.apps.apps.filter((app) => app.usingBlindSig);

export const selectExpireableBlindSigApps = createSelector(
  selectBlindSigApps,
  (apps) =>
    apps.filter((app) => app.verificationExpirationLength && !app.testing),
);

export const selectSponsoringStep = (state: RootState) =>
  state.apps.sponsoringStep;

export const selectSponsoringStepText = (state: RootState) =>
  state.apps.sponsoringStepText;

export const selectLinkingAppInfo = (state: RootState) =>
  state.apps.linkingAppInfo;

export const selectLinkingAppStartTime = (state: RootState) =>
  state.apps.linkingAppStartTime;

export const selectSponsorOperationHash = (state: RootState) =>
  state.apps.sponsorOperationHash;

// Export reducer
export default appsSlice.reducer;
