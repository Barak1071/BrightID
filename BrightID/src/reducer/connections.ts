import { mergeRight } from 'ramda';
import { keyBy } from 'lodash';
import {
  SET_CONNECTIONS,
  CONNECTIONS_SORT,
  UPDATE_CONNECTIONS,
  DELETE_CONNECTION,
  ADD_CONNECTION,
  SET_CONNECTIONS_SEARCH,
  SET_CONNECTIONS_SEARCH_OPEN,
  HYDRATE_CONNECTIONS,
  RESET_STORE,
  FLAG_AND_HIDE_CONNECTION,
  SHOW_CONNECTION,
  STALE_CONNECTION,
  SET_CONNECTION_LEVEL,
  SET_FILTERS,
} from '@/actions';
import { connection_levels } from '@/utils/constants';

export const initialState = {
  connections: [],
  connectionsSort: '',
  searchParam: '',
  searchOpen: false,
  filters: [
    connection_levels.SUSPICIOUS,
    connection_levels.JUST_MET,
    connection_levels.ALREADY_KNOWN,
    connection_levels.RECOVERY,
  ],
};

export const reducer = (
  state: ConnectionsState = initialState,
  action: action,
) => {
  switch (action.type) {
    case SET_CONNECTIONS: {
      return {
        ...state,
        connections: action.connections.slice(0),
      };
    }
    case UPDATE_CONNECTIONS: {
      const connectionsMap = keyBy(action.connections, 'id');
      return {
        ...state,
        connections: state.connections.map<connection>((conn: connection) => {
          const updatedConn = connectionsMap[conn.id];
          if (!updatedConn) {
            if (conn.status === 'verified') conn.status = 'deleted';
            return conn;
          } else {
            conn.status = 'verified';
            return mergeRight(conn, updatedConn);
          }
        }),
      };
    }
    case ADD_CONNECTION: {
      if (!action.connection.id) return state;
      const removeExisting = ({ id }: connection) =>
        id !== action.connection.id;
      console.log('adding connection', action.connection.id);
      return {
        ...state,
        connections: [
          action.connection,
          ...state.connections.filter(removeExisting),
        ],
      };
    }
    case DELETE_CONNECTION: {
      const connections: connection[] = state.connections.filter(
        (conn: connection) => conn.id !== action.id,
      );
      return {
        ...state,
        connections,
      };
    }
    case FLAG_AND_HIDE_CONNECTION: {
      const connections: connection[] = state.connections.map(
        (conn: connection) => {
          if (conn.id === action.id) {
            conn.status = 'hidden';
            conn.hiddenFlag = action.flag;
          }
          return conn;
        },
      );
      return {
        ...state,
        connections,
      };
    }
    case SHOW_CONNECTION: {
      const connections: connection[] = state.connections.map(
        (conn: connection) => {
          if (conn.id === action.id) {
            conn.status = 'verified';
          }
          return conn;
        },
      );
      return {
        ...state,
        connections,
      };
    }
    case STALE_CONNECTION: {
      const connections: connection[] = state.connections.map(
        (conn: connection) => {
          if (conn.id === action.id) {
            conn.status = 'stale';
          }
          return conn;
        },
      );
      return {
        ...state,
        connections,
      };
    }
    case CONNECTIONS_SORT: {
      return {
        ...state,
        connectionsSort: action.connectionsSort,
      };
    }
    case SET_CONNECTIONS_SEARCH: {
      return {
        ...state,
        searchParam: action.searchParam,
      };
    }
    case SET_CONNECTIONS_SEARCH_OPEN: {
      return {
        ...state,
        searchOpen: action.searchOpen,
      };
    }
    case SET_CONNECTION_LEVEL: {
      const connections: connection[] = state.connections.map(
        (conn: connection) => {
          if (conn.id === action.id) {
            conn.level = action.level;
          }
          return conn;
        },
      );
      return {
        ...state,
        connections,
      };
    }
    case SET_FILTERS: {
      return {
        ...state,
        filters: action.filters,
      };
    }
    case HYDRATE_CONNECTIONS: {
      if (!action.data.connections) return state;

      return { ...action.data };
    }
    case RESET_STORE: {
      return { ...initialState };
    }
    default: {
      return state;
    }
  }
};

export default reducer;