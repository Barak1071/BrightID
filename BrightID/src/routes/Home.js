import * as React from 'react';
import { Image, TouchableOpacity, View } from 'react-native';
import { useSelector } from 'react-redux';
import { INVITE_ACTIVE, DEVICE_LARGE } from '@/utils/constants';
import { createStackNavigator } from '@react-navigation/stack';
import Material from 'react-native-vector-icons/MaterialCommunityIcons';
import HomeScreen from '@/components/HomeScreen';
import RecoveringConnectionScreen from '@/components/Recovery/RecoveringConnectionScreen';

import { navigate } from '@/NavigationService';

const Stack = createStackNavigator();

const homeScreenOptions = (notificationCount) => ({
  headerTitle: () => (
    <Image
      source={require('@/static/brightid-final.png')}
      accessible={true}
      accessibilityLabel="Home Header Logo"
      resizeMode="contain"
      style={{ width: DEVICE_LARGE ? 104 : 85 }}
    />
  ),
  headerLeft: () => null,
  headerRight: () => (
    <TouchableOpacity
      style={{ marginRight: 25 }}
      onPress={() => {
        navigate('Notifications');
      }}
    >
      <Material name="bell" size={DEVICE_LARGE ? 28 : 23} color="#000" />
      {notificationCount ? (
        <View
          style={{
            backgroundColor: '#ED1B24',
            width: 9,
            height: 9,
            borderRadius: 5,
            position: 'absolute',
            top: 5,
            left: 17,
          }}
        />
      ) : null}
    </TouchableOpacity>
  ),
  headerStyle: {
    height: DEVICE_LARGE ? 80 : 70,
    shadowRadius: 0,
    shadowOffset: {
      height: 0,
    },
    elevation: 0,
  },
  headerTitleAlign: 'center',
});

const recoveringConnectionOptions = {
  title: 'Account Recovery',
  headerShown: DEVICE_LARGE,
};

const Home = () => {
  const notificationCount = useSelector(
    ({ user: { notifications }, groups: { invites } }) =>
      notifications?.length +
      invites?.filter((invite) => invite.state === INVITE_ACTIVE)?.length,
  );
  return (
    <>
      <Stack.Screen
        name="Home"
        component={HomeScreen}
        options={homeScreenOptions(notificationCount)}
      />
      <Stack.Screen
        name="RecoveringConnection"
        component={RecoveringConnectionScreen}
        options={recoveringConnectionOptions}
      />
    </>
  );
};

export default Home;
