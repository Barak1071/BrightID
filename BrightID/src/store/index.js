// @flow

import AsyncStorage from '@react-native-community/async-storage';
import thunkMiddleware from 'redux-thunk';
import { applyMiddleware, createStore, compose } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import reducer from '@/reducer';
import { migrate } from './migrations';
import {
  groupsTransformer,
  qrDataTransformer,
  userTransformer,
  connectionsTransformer,
} from './transform';

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
  transforms: [
    userTransformer,
    groupsTransformer,
    qrDataTransformer,
    connectionsTransformer,
  ],
  version: 6,
  migrate,
  timeout: null,
};

const persistedReducer = persistReducer(persistConfig, reducer);

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(
  persistedReducer,
  composeEnhancers(applyMiddleware(thunkMiddleware)),
);

export const persistor = persistStore(store);

export default store;
