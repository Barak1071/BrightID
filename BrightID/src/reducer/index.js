import { combineReducers } from 'redux';
import apps from './apps';
import channels from '../components/PendingConnectionsScreens/channelSlice';
import connections from './connections';
import groups from './groups';
import notifications from './notifications';
import operations from './operations';
import pendingConnections from '../components/PendingConnectionsScreens/pendingConnectionSlice';
import recoveryData from './recoveryData';
import user from './user';

export default combineReducers({
  apps,
  channels,
  connections,
  groups,
  operations,
  pendingConnections,
  recoveryData,
  notifications,
  user,
});
