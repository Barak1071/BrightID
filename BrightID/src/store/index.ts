import { persistStore, persistReducer } from 'redux-persist';
import {
  combineReducers,
  configureStore,
  getDefaultMiddleware,
} from '@reduxjs/toolkit';
import reducers from '@/reducer';
import FsStorage from './storage/fsStorageAdapter';
import KeychainStorage from './storage/keychainAdapter';
import getStoredState from './getStoredState';
import { appsMigrate } from './migrations/apps';
import { connectionsMigrate } from './migrations/connections';
import { ChannelsTransform, RecoveryDataTransform } from '@/store/transforms';

// update this in async migrate if changed to prevent require cycle

const version = 11;

const fsPersistConfig = {
  storage: FsStorage,
  timeout: 0,
  // debug: __DEV__,
  version,
  getStoredState,
};

const appsPersistConfig = {
  ...fsPersistConfig,
  key: 'apps',
  migrate: appsMigrate,
  blacklist: [
    'sigsUpdating',
    'sponsoringStep',
    'linkingAppStartTime',
    'linkingAppInfo',
  ],
};

const connectionsPersistConfig = {
  ...fsPersistConfig,
  key: 'connections',
  migrate: connectionsMigrate,
  blacklist: ['searchParam', 'searchOpen', 'filters'],
};

const groupsPersistConfig = {
  ...fsPersistConfig,
  key: 'groups',
  blacklist: ['searchParam', 'searchOpen', 'newGroupInvitees'],
};

const notificationsPersistConfig = {
  ...fsPersistConfig,
  key: 'notifications',
  blacklist: ['activeNotification', 'sessionNotifications'],
};

const userPersistConfig = {
  ...fsPersistConfig,
  key: 'user',
  blacklist: ['searchParam', 'migrated'],
};

const tasksPersistConfig = {
  ...fsPersistConfig,
  key: 'tasks',
};

const devicesPersistConfig = {
  ...fsPersistConfig,
  key: 'devices',
};

const socialMediaPersistConfig = {
  ...fsPersistConfig,
  key: 'socialMedia',
};

const settingsPersistConfig = {
  ...fsPersistConfig,
  key: 'settings',
  blacklist: ['baseUrl'],
};

const operationsPersistConfig = {
  ...fsPersistConfig,
  key: 'operations',
};

const keypairPersistConfig = {
  key: 'keypair',
  storage: KeychainStorage,
  timeout: 0,
  // debug: __DEV__,
  version,
  getStoredState,
  serialize: false,
  deserialize: false,
};

const channelsPersistConfig = {
  ...fsPersistConfig,
  key: 'channels',
  transforms: [ChannelsTransform],
};

const recoveryDataPersistConfig = {
  ...fsPersistConfig,
  key: 'recoveryData',
  transforms: [RecoveryDataTransform],
};

const rootReducer = combineReducers({
  ...reducers,
  apps: persistReducer(
    appsPersistConfig,
    reducers.apps,
  ) as typeof reducers.apps,
  connections: persistReducer(connectionsPersistConfig, reducers.connections),
  groups: persistReducer(groupsPersistConfig, reducers.groups),
  keypair: persistReducer(keypairPersistConfig, reducers.keypair),
  notifications: persistReducer(
    notificationsPersistConfig,
    reducers.notifications,
  ),
  socialMedia: persistReducer(socialMediaPersistConfig, reducers.socialMedia),
  tasks: persistReducer(tasksPersistConfig, reducers.tasks),
  user: persistReducer(userPersistConfig, reducers.user),
  settings: persistReducer(settingsPersistConfig, reducers.settings),
  operations: persistReducer(operationsPersistConfig, reducers.operations),
  devices: persistReducer(devicesPersistConfig, reducers.devices),
  channels: persistReducer(
    channelsPersistConfig,
    reducers.channels,
  ) as typeof reducers.channels,
  recoveryData: persistReducer(
    recoveryDataPersistConfig,
    reducers.recoveryData,
  ) as typeof reducers.recoveryData,
});

export const store = configureStore({
  reducer: rootReducer,
  middleware: getDefaultMiddleware({
    // We have a bunch of non-serializable data like secret key etc.
    // TODO For now disabled completely. Revisit later for fine-grained configuration.
    serializableCheck: false,
    immutableCheck: false,
  }),
});

export const persistor = persistStore(store);

export default store;
