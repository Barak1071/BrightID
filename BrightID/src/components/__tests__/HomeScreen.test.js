import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { Text, View } from 'react-native';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { configure, shallow } from 'enzyme';
import { initialState } from '../../reducer';
import store from '../../store';
import {
  setUserData,
  setUserScore,
  setGroupsCount,
  setConnections,
} from '../../actions';
import Home, { HomeScreen } from '../HomeScreen';

configure({ adapter: new Adapter() });

describe('homescreen', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Home />
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('name is updated', () => {
    // updating redux store should update homescreen component

    const main1 = store.getState().main;
    const wrapper1 = shallow(<HomeScreen {...main1} />);

    expect(wrapper1.find('#name').children()).toHaveLength(0);

    store.dispatch(
      setUserData({
        name: 'Shiva',
      }),
    );

    const main2 = store.getState().main;
    const wrapper2 = shallow(<HomeScreen {...main2} />);

    expect(wrapper2.find('#name').children()).toHaveLength(1);

    expect(
      wrapper2
        .find('#name')
        .children()
        .text(),
    ).toBe('Shiva');
  });

  test('score is updated', () => {
    // updating redux store should update homescreen component

    const main1 = store.getState().main;
    const wrapper1 = shallow(<HomeScreen {...main1} />);

    expect(wrapper1.find('#score').children()).toHaveLength(1);

    expect(
      wrapper1
        .find('#score')
        .children()
        .text(),
    ).toBe('% Trusted');

    store.dispatch(setUserScore(99.8));

    const main2 = store.getState().main;
    const wrapper2 = shallow(<HomeScreen {...main2} />);

    expect(wrapper2.find('#score').children()).toHaveLength(2);

    expect(
      wrapper2
        .find('#score')
        .children()
        .first()
        .text(),
    ).toBe('99.8');

    expect(
      wrapper2
        .find('#score')
        .children()
        .last()
        .text(),
    ).toBe('% Trusted');
  });

  test('connectionsCount is updated', () => {
    // updating redux store should update homescreen component

    const main1 = store.getState().main;
    const wrapper1 = shallow(<HomeScreen {...main1} />);

    expect(wrapper1.find('#connectionsCount').children()).toHaveLength(1);

    expect(
      wrapper1
        .find('#connectionsCount')
        .children()
        .text(),
    ).toBe('1');

    store.dispatch(setConnections([...Array(8)]));

    const main2 = store.getState().main;
    const wrapper2 = shallow(<HomeScreen {...main2} />);

    expect(wrapper2.find('#connectionsCount').children()).toHaveLength(1);

    expect(
      wrapper2
        .find('#connectionsCount')
        .children()
        .text(),
    ).toBe('8');
  });

  test('groupsCount is updated', () => {
    // updating redux store should update homescreen component

    const main1 = store.getState().main;
    const wrapper1 = shallow(<HomeScreen {...main1} />);

    expect(wrapper1.find('#groupsCount').children()).toHaveLength(0);

    store.dispatch(setGroupsCount(10));

    const main2 = store.getState().main;
    const wrapper2 = shallow(<HomeScreen {...main2} />);

    expect(wrapper2.find('#groupsCount').children()).toHaveLength(1);

    expect(
      wrapper2
        .find('#groupsCount')
        .children()
        .text(),
    ).toBe('10');
  });
});
