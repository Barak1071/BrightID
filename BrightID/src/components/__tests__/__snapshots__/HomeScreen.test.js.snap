// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homescreen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "marginTop": 24,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 2,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/static/default_avatar.jpg",
            }
          }
          style={
            Object {
              "borderRadius": 71,
              "height": 142,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowRadius": 4,
              "width": 142,
            }
          }
        />
      </View>
      <Text
        id="name"
        style={
          Object {
            "fontFamily": "ApexNew-Book",
            "fontSize": 30,
            "marginTop": 8,
            "shadowColor": "rgba(0,0,0,0.32)",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowRadius": 4,
            "textAlign": "center",
          }
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#e3e1e1",
          "borderBottomWidth": 1,
          "borderTopColor": "#e3e1e1",
          "borderTopWidth": 1,
          "marginTop": 9,
          "paddingBottom": 7,
          "paddingTop": 7,
          "width": "80%",
        }
      }
    >
      <Text
        id="score"
        style={
          Object {
            "color": "green",
            "fontFamily": "ApexNew-Book",
            "fontSize": 18,
            "textAlign": "center",
          }
        }
      >
        
        % Trusted
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "marginTop": 12,
          "width": "75%",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          id="connectionsCount"
          style={
            Object {
              "fontFamily": "ApexNew-Book",
              "fontSize": 22,
              "textAlign": "center",
            }
          }
        >
          1
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "ApexNew-Book",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Connections
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          id="groupsCount"
          style={
            Object {
              "fontFamily": "ApexNew-Book",
              "fontSize": 22,
              "textAlign": "center",
            }
          }
        >
          0
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "ApexNew-Book",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Groups
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#f9f9f9",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 17,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4990e2",
            "borderRadius": 69,
            "elevation": 1,
            "height": 138,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 20,
              "width": 20,
            },
            "shadowRadius": 20,
            "width": 138,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontFamily": "ApexNew-Book",
              "fontSize": 16,
              "marginBottom": 5,
              "marginTop": 10,
            }
          }
        >
          CONNECT
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 26,
              },
              undefined,
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 63,
        "justifyContent": "space-evenly",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 32,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "ApexNew-Book",
              "fontSize": 12,
            }
          }
        >
          Home
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 32,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "ApexNew-Book",
              "fontSize": 12,
            }
          }
        >
          Connections
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 32,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "ApexNew-Book",
              "fontSize": 12,
            }
          }
        >
          Groups
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 32,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "ApexNew-Book",
              "fontSize": 12,
            }
          }
        >
          Notifications
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 32,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "ApexNew-Book",
              "fontSize": 12,
            }
          }
        >
          Apps
        </Text>
      </View>
    </View>
  </View>
</View>
`;
